src/Data/Text/AhoCorasick/Searcher.hs

```diff
@@ -152,5 +154,35 @@ containsAny !searcher !text =
     -- On the first match, return True immediately.
     f _acc _match = Aho.Done True
   in case caseSensitivity searcher of
-    CaseSensitive  -> Aho.runText False f (automaton searcher) text
-    IgnoreCase      -> Aho.runLower False f (automaton searcher) text
+    CaseSensitive -> Aho.runText False f (automaton searcher) text
+    IgnoreCase   -> Aho.runLower False f (automaton searcher) text
+
+-- | Build a 'Searcher' that returns the needle's index in the needle list when it matches.
+buildNeedleIdSearcher :: CaseSensitivity -> [Text] -> Searcher Int
+buildNeedleIdSearcher !case_ !ns =
+  buildWithValues case_ $ zip ns [0.. length ns - 1]
+
+-- | Returns whether the haystack contains all of the needles.
+-- This function expects the passed 'Searcher' to be constructed using 'buildNeedleIdAutomaton'.
+containsAll :: Searcher Int -> Text -> Bool
+containsAll !searcher !haystack =
+  let
+    initial = buildIntSetFromTo 0 $ numNeedles searcher
+    ac = automaton searcher
+
+    f !acc (Aho.Match _index !needleId)
+      | IS.null acc' = Aho.Done acc'
+      | otherwise = Aho.Step acc'
+      where
+        !acc' = IS.delete needleId acc
+
+  in IS.null $ case caseSensitivity searcher of
+    CaseSensitive -> Aho.runText initial f ac haystack
+    IgnoreCase   -> Aho.runLower initial f ac haystack
+
+-- | Build an 'IntSet' containing the given range.
+-- The resulting set contains all 'Int's greater than or equal to the first and smaller than the second.
+buildIntSetFromTo :: Int -> Int -> IS.IntSet
+buildIntSetFromTo = go IS.empty
+  where go !acc !lo !hi | lo >= hi  = acc
+                        | otherwise = go (IS.insert lo acc) (lo + 1) hi
```

Comment:
I just measured it, it is not. I introduced this function because `fromAscList` is marked as `NOINLINE` [1] and the Core for the code using it allocated a bunch of conses. However on our benchmark data set it does not make a difference, so I'd revert that change and go with the more obvious `fromAscList`.

[1][https://hackage.haskell.org/package/containers-0.6.5.1/docs/src/Data.IntSet.Internal.html#fromAscList]